### DISCUSSION TITLE
Potential issue with string example

### DISCUSSION THREAD

**Discussion by max-sixty** [pest-parser/pest#645] [General]
Currently the [literals section of the book](https://pest.rs/book/examples/rust/literals.html) gives this as a string example:

```
string = { "\"" ~ (raw_string | escape)* ~ "\"" }
```

Is my understanding correct that this will parse `" USA"` as `USA`, discarding the initial ` `, because the rule isn't atomic?

I'm having this issue myself. I'm getting around it by making the rule atomic, but it seems rules can't be atomic & silent, and so the parse tree contains the `"`s, which requires a bit more work downstream.

Thanks!

**Comment by max-sixty** [comment]
Re the transfer: I think this is a bug in the documentation rather than a discussion question

**Reply by tomtau** [reply]
Oh ok, here are two follow up issues:
1. **pest**: there is a similar issue that was opened about it: https://github.com/pest-parser/pest/issues/520 (and see: https://github.com/pest-parser/pest/issues/271#issuecomment-883207393 )
2. **book**: I re-posted your original issue in the book repository: https://github.com/pest-parser/book/issues/23 so that the current behaviour could be better explained in the book
